packages/components/src/advanced-filters/index.js

```diff
@@ -117,11 +117,12 @@ class AdvancedFilters extends Component {
 		onAdvancedFilterAction( 'match', { match } );
 	}
 
-	onFilterChange( index, property, value ) {
+	onFilterChange( index, property, value, shouldResetValue = false ) {
 		const newActiveFilters = [ ...this.state.activeFilters ];
 		newActiveFilters[ index ] = {
 			...newActiveFilters[ index ],
 			[ property ]: value,
+			...( shouldResetValue === true ? { value: null } : {} ),
```

Comment:
Hmm I am not a huge fan of adding this extra parameter, the `onFilterChange` functions seems to only be used for updating the value, so adding a `shouldResetValue` for the one case where we don't update the value seems confusing.
I would be good with going to different routes for this:
- Add a separate `onRuleChange` function that handles the reset value, although this is very specific (so maybe not a great option)
- Update the `onFilterChange` function to pass in a updated filterValue object instead:
```
onFilterChange( index, updatedFilterValue ) {
  const newActiveFilters = [ ...this.state.activeFilters ];
  newActiveFilters[ index ] = {
			...newActiveFilters[ index ],
            ...updatedFilterValue
  }
 this.setState( { activeFilters: newActiveFilters } );
}
```
This way it is more explicit in the date-filter what you are doing ( updating the role and value )

Reply:
Hey @louwie17, thanks for the review!

> Hmm I am not a huge fan of adding this extra parameter

I agree. I went with this approach as it was low touch and didn't require too much refactoring.

> Update the onFilterChange function to pass in a updated filterValue object ... This way it is more explicit in the date-filter what you are doing ( updating the role and value )

This is definitely a better approach but would require refactoring the existing filter components (e.g. `AttributeFilter`, `NumberFilter`, `SearchFilter`, etc.)

Ultimately, this probably isn't too difficult but given this is my first PR to this repo I didn't want to get too adventurist.

> This would require a potential bigger refactor, so I am fine if this happens in a separate PR and we can merge this as is (unless you feel inclined to make the changes as part of this PR).

Cool thanks. In that case, I might go ahead with current approach and circle back with the aim of improving this in the future. I'll create a new GitHub issue for that now.